src/auro-input.js

```diff
@@ -106,16 +114,140 @@ export default class AuroInput extends BaseInput {
     return null
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (this.format) {
+      var cleave = new Cleave(this, this.format);
+    }
+  }
+
+  getCreditCardIcon() {
+    return this.icon && this.format && this.format.hasOwnProperty('creditCard') ?
+      html`
+        <div class="creditCard-icon">
+         ${this.generateCreditCardIconHtml()}
+        </div>
+      ` :
+      html``
+  }
+
+  generateCreditCardIconHtml() {
+    let icon; //
+
+    if (!this.value) {
+      icon = creditCard;
+    } else {
+      const v = this.value.trim();
+
+      if (v.length === 0) {
+        icon = creditCard;
+      }
+
+      if (v.length === 1) {
+        if (parseInt(v) === 4) {
+          // Visa
+          icon = creditCardVisa;
+        }
+      }
+
+      if (v.length === 2) {
+        const val = parseInt(v.substring(0, 2));
+
+        switch(val) {
+          case 34: case 37:
+            // American Express
+            icon = creditCardAmex;
+            break;
+          case 35:
+            // JCB card
+            icon = creditCard;
+            break;
+          case 22: case 23: case 24: case 25: case 26: case 27: case 51: case 52: case 53: case 54: case 55:
+            // Master Card
+            icon = creditCardMastercard;
+            break;
+          case 65:
+            // Discover Card
+            icon = creditCardDiscover;
+            break;
+        }
+      }
+
+      if (v.length === 3) {
+        const val = parseInt(v.substring(0, 3));
+
+        switch(val) {
+          case 300: case 301: case 302: case 303: case 304: case 305: case 309:
+            // Diners Club
+            icon = creditCard;
+            break;
+          case 644: case 645: case 646: case 647: case 648: case 649:
+            // Discover Card
+            icon = creditCardDiscover;
+            break;
+        }
+      }
+
+      if (v.length === 4) {
+        const val = parseInt(v.substring(0, 4));
+
+        switch(val) {
+          case 1800: case 2131:
+            // JCB card
+            icon = creditCard;
+            break;
+          case 6011:
+            // Discover card
+            icon = creditCardDiscover;
+            break;
+        }
+      }
+    }
+
+    if (icon) {
```

Comment:
This could use comments or possibly be higher up in the code. It took me a long time to find this as the hook for showing the icon or not. There is a lot of code for this feature so some comments on how this all ties together would be very helpful.

Reply:
Added private comment